assignments/002-unix-intro/unix-intro.sh

```diff
@@ -0,0 +1,593 @@
+(base) [zhimei@colostate.edu@login11 nice_dir_name]$ echo "A string in a fil
+e" > nice_file_name
+(base) [zhimei@colostate.edu@login11 nice_dir_name]$ cat nice_file_name
+A string in a file
+
+#### Ex. 20: Catenate the contents of `nice_dir_name/nice_file_name` to _stdout_
+
+(base) [zhimei@colostate.edu@login11 002-unix-intro]$ cat nice_dir_name
+cat: nice_dir_name: Is a directory
+(base) [zhimei@colostate.edu@login11 nice_dir_name]$ cat nice_file_name
+A string in a file
+
+#### Ex. 21: Create a directory called `bad directory name with spaces`, and then,
+####         with redirection, create a file inside it, named 
+####         `bad file name with spaces` whose contents are the string
+####         `Whoa! No spaces, please!`. Use backslash escaping instead of quoting
+
+(base) [zhimei@colostate.edu@login11 002-unix-intro]$ mkdir "bad directory name with spaces"
+(base) [zhimei@colostate.edu@login11 002-unix-intro]$ ls
+bad directory name with spaces  fastq          sam
+bam                             metadata       silly
+count_files.sh                  nice_dir_name  unix-intro-TEMPLATE.sh
+
+(base) [zhimei@colostate.edu@login11 bad] touch "bad file name with spaces"
+(base) [zhimei@colostate.edu@login11 bad] cat  "Whoa! No spaces, please!" > bad file name with spaces
+
+#### Ex. 22: Catenate the contents of 
+####         `bad directory name with spaces/bad file name with spaces` to stdout.
+
+(base) [zhimei@colostate.edu@login11 002-unix-intro]$ cat "bad directory name with spaces"
+cat: bad directory name with spaces: Is a directory
+(base) [zhimei@colostate.edu@login11 bad directory name with spaces]$ cat "bad file name with spaces"
+cat: bad file name with spaces: Is a file
+
+
+#### Ex. 23: List the directories (just the names, not the contents (`-d`)) of
+####         `bad directory name with spaces` and `nice_dir_name`.
+
+(base) [zhimei@colostate.edu@login11 002-unix-intro]$ ls -d {"bad directory name with spaces",nice_dir_name}
+bad directory name with spaces  nice_dir_name
+
+#### Ex. 24: Once you are convinced that you can list the directories
+####         correctly in your terminal, recall that `ls` command (i.e. get it back on the
+####         command line using the up arrow on your keyboard) and replace `ls` with
+####         `rm -r` to remove them. BE CAREFUL with the rm command!
+
+(base) [zhimei@colostate.edu@login11 002-unix-intro]$ rm -r {"bad directory name with spaces",nice_dir_name}
+
+#### Ex. 25: Why is it a good idea to `ls` things before removing them?
+
+This is because is to double check what it is we are removing before we remove them since, 
+there is no way to get them back once they have been removed. 
+
+#### Ex. 26: If you had just typed `rm -r bad directory name with spaces` 
+####         what would Unix have tried to remove? (No command to type here,
+####         just answer the question).
+
+Unix will try to remove a directory called bad instead of "bad directory name with spaces". 
```

Comment:
It would have tried to remove more than just `bad`. Also would have tried to remove directories named:
- directory
- name
- with
- spaces